core-service/src/main/java/com/crmfoodestablishment/coreservice/service/MenuService.java

```diff
@@ -14,36 +16,49 @@ public class MenuService {
 
     private final MenuRepository menuRepository;
 
+    private final MenuMapper menuMapper;
+
     private Menu checkIfMenuExist(Integer id) {
         return menuRepository.findById(id)
                 .orElseThrow(() -> new MenuNotFoundException("No menu found for this id"));
+
     }
 
-    public Menu addMenu(Menu menu) {
-        boolean idExist = menuRepository.existsById(menu.getId());
+    public Integer addMenu(MenuDto menuDto) {
+        boolean idExist = menuRepository.existsById(menuDto.getId());
```

Comment:
As polina said consider creating something like CreateMenuRequestPayload, because id is not input data. Due to generating id in DB that line is just unreasonable

Reply:
Understood, thanks.